consensus/polybft/consensus_runtime.go

```diff
@@ -115,22 +117,38 @@ type consensusRuntime struct {
 	// activeValidatorFlag indicates whether the given node is amongst currently active validator set
 	activeValidatorFlag uint32
 
+	// checkpointsOffset represents offset between checkpoint blocks (applicable only for non-epoch ending blocks)
+	checkpointsOffset uint64
+
+	// checkpointManager represents abstraction for checkpoint submission
+	checkpointManager *checkpointManager
+
+	// logger instance
 	logger hcf.Logger
 }
 
 // newConsensusRuntime creates and starts a new consensus runtime instance with event tracking
 func newConsensusRuntime(log hcf.Logger, config *runtimeConfig) (*consensusRuntime, error) {
+	blockTimeInMillis := config.PolyBFTConfig.BlockTime.Milliseconds()
 	runtime := &consensusRuntime{
-		state:  config.State,
-		config: config,
-		logger: log.Named("consensus_runtime"),
+		state:             config.State,
+		config:            config,
+		logger:            log.Named("consensus_runtime"),
+		checkpointsOffset: uint64(checkpointTimeInterval.Milliseconds() / blockTimeInMillis),
 	}
 
 	if runtime.IsBridgeEnabled() {
 		err := runtime.startEventTracker()
 		if err != nil {
 			return nil, err
 		}
+
+		runtime.checkpointManager = &checkpointManager{
```

Comment:
Can I suggest making a constructor function in `checkpointManager`, which will receive `rootchainInteractor` as a parameter. If its nil, we can use this `defaultRootchainInteractor` as an implementation. The constructor will also receive blockTime as a parameter, and if you move the default checkpoint submit time, you can calculate that `offset` in the constructor. That way you have that logic encapsulated in `checkpointManager` without the need that outside source knows all this.

Reply:
Indeed!

Reply:
Fixed [6a2a52b](https://github.com/0xPolygon/polygon-edge/pull/861/commits/6a2a52ba4577cd8c48852e9273e77ae3fc5492fe)